rust/lance/src/dataset/write.rs

```diff
@@ -112,8 +122,31 @@ pub fn reader_to_stream(
 /// NOTE: the fragments have not yet been assigned an ID. That must be done
 /// by the caller. This is so this function can be called in parallel, and the
 /// IDs can be assigned after writing is complete.
-#[instrument(skip_all)]
 pub async fn write_fragments(
+    dataset_uri: &str,
+    data: impl RecordBatchReader + Send + 'static,
```

Comment:
Is the soft-limit being applid after a RecordBatch is written?

Need to advise users don't write a `pyarrow.Table` with single batch?

Reply:
> Need to advise users don't write a pyarrow.Table with single batch?

We break up the input into groups, so the limit is being applied after each group (default 1024 rows).

I mention this in the Python API:

> This limit is checked at the group level, so if your group size is large then this can overshoot the limit.